cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/task/LoadMonitorTaskRunner.java

```diff
@@ -262,21 +264,29 @@ public void shutdown() {
   /**
    * Pause the scheduled sampling tasks..
    */
-  public void pauseSampling() {
+  public synchronized void pauseSampling() {
     if (_state.get() != PAUSED && !_state.compareAndSet(RUNNING, PAUSED)) {
       throw new IllegalStateException("Cannot pause the load monitor because it is in " + _state.get() + " state.");
     }
+    _awaitExecution = true;
```

Comment:
If we throw an exception before setting this variable (e.g. this will happen when in `SAMPLING` state), the sampler would be unable set `_awaitExecution` to `true`. Also, do we need to set it if we can successfully set state to `PAUSED`?
Hence, should we move this assignment to `true` inside the `if`, just before the exception and reset it if we can successfully pause? -- i.e.

```
    if (_state.get() != PAUSED && !_state.compareAndSet(RUNNING, PAUSED)) {
      _awaitingPauseSampling = true;
      throw new IllegalStateException("Cannot pause the load monitor because it is in " + _state.get() + " state.");
    } else {
      _awaitingPauseSampling = false;
    }
```